Assets/Scripts/HealthbarController.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class HealthbarController : MonoBehaviour
+{
+    Vector3 localScale;
+    // Start is called before the first frame update
+    void Start()
+    {
+        localScale = transform.localScale;
+    }
+    // Update is called once per frame
+    void Update()
+    {
+        localScale.y = transform.parent.gameObject.GetComponent<CharacterController>().getHpScale();
```

Comment:
Update() is called once every frame, which this does not need to do. 

If you put this code in a function that is called by `CharacterController.TakeDamage()`, you can reduce the number of times this code gets run and keep resources free.

This also would allow you to pass data from `CharacterController` in as arguments so you don't have to make a reference to it.

Reply:
(this also would allow you to not have both `hp` and `hpScale` in `CharacterController`, but instead modify the scale in this script, further reducing cross-referencing.)

Reply:
Done, good catch